src/components/CheckBox/CheckBox.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import style from "./CheckBox.module.scss";
+
+export type CheckBoxProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  "onChange"
+> & {
+  /** Вызывается при клике на чекбокс */
+  onChange: (checked: boolean) => void;
+};
+
+const CheckBox: React.FC<CheckBoxProps> = ({ onChange, ...props }) => {
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onChange(event.target.checked);
+  };
+
+  const getClassName = () => {
+    if (props.disabled) {
+      return props.checked
+        ? style.checkboxDisabledChecked
+        : style.checkboxDisabled;
+    }
+    return props.checked ? style.checkboxChecked : style.checkbox;
+  };
+
+  return (
+    <label className={getClassName()}>
+      <input type="checkbox" onChange={handleChange} {...props} />
+    </label>
+  );
```

Comment:
[!] У тебя установлен пакет "classnames", поэтому для такой логики можно использовать его

```suggestion
  return (
    <label className={cn({
      [style.checkboxDisabledChecked]: props.disabled && props.checked,
      [style.checkboxDisabled]: props.disabled && !props.checked,
      [style.checkboxChecked]: props.checked && !props.disabled,
      [style.checkbox]: !props.checked && !props.disabled,
    })}>
      <input type="checkbox" onChange={handleChange} {...props} />
    </label>
  );
```

Но вообще как-то слишком много классов. Обычно достаточно двух css-классов модификаторов для disabled состояния и для checked))